internal/operators/restpath_test.go

```diff
@@ -25,6 +25,33 @@ func TestRestPath(t *testing.T) {
 		t.Errorf("Expected %s to match %s", exp, path)
 	}
 	if tx.Variables().ArgsPath().Get("id")[0] != "123" {
-		t.Errorf("Expected 123, got %s", tx.Variables().ArgsPath().Get("id"))
+		t.Errorf("Expected id to be 123, got %s", tx.Variables().ArgsPath().Get("id"))
+	}
+}
+
+func TestRestPathQueryShouldNotBeGreedy(t *testing.T) {
+	waf := corazawaf.NewWAF()
+	tx := waf.NewTransaction()
+
+	exp := "/some-random/url/{id}"
+	testCases := map[string]string{
+		"/some-random/url/123?q=test": "123", // ?q=test is query info
+		"/some-random/url/456/test":   "456", // /test is extra path info
+	}
```

Comment:
Thanks for this. Would you mind adding more tests, with sub variables?

E.g. `/some-random/username/{uuid}/name/{name}/test`, etc.

Reply:
Good call. Sub variables did work, but while I was adding tests, I went ahead and added additional restpath tests in general as well. This actually uncovered a second existing issue, where empty path elements at the end were not correctly preventing a match. See the last test `TestRestPathShouldNotMatchOnEmptyPathElement` for a previously failing case that was also fixed in this PR.